### PR TITLE
Add first batch of type hints and pydantic to the library

### PR DESCRIPTION
**Pull request by alesgenova**

### CODE REVIEW

File: examples/operators/build.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+root_dir=$(git rev-parse --show-toplevel)
+
+# First, make sure that oremda is built
+$root_dir/docker/oremda/build.sh
+
+# Next, build all the example directories
+for name in "${directories[@]}"
+do
+  bash $script_dir/$name/build.sh
+done
```
Comment (cjh1) [inline examples/operators/build.sh:25]: Are these changes related to type hinting?

**Review by cjh1 — APPROVED** [review]
Very nice! Just one inline question, seems like there are some other changes mixed in here?

File: oremda/operator.py [hunk 2]
```diff
@@ -53,48 +56,65 @@ def operate(self, info):
...
+            name = _name
 
         @wraps(func)
-        def kernel(self, *args, **kwargs):
+        def kernel(_, *args, **kwargs):
             # Remove self so the caller does not need to add it
             return func(*args, **kwargs)
 
-        class_name = f'{name.capitalize()}Operator'
+        class_name = f'{name.capitalize()}Operator'
```
Comment (psavery) [inline oremda/operator.py:117]: Can you set up your text editor to automatically remove trailing whitespace when you save? 

[Here is an example for visual studio code](https://stackoverflow.com/a/53663494).

**Review by psavery — APPROVED** [review]
LGTM

File: oremda/typing/__init__.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    image: str
+    params: JSONType
+
+class EdgeJSON(BaseModel):
+    type: PortType = Field(...)
+    start: PortJSON = Field(..., alias='from')
+    stop: PortJSON = Field(..., alias='to')
+
+class PipelineJSON(BaseModel):
+    nodes: Sequence[NodeJSON] = []
```
Comment (psavery) [inline oremda/typing/__init__.py:87]: It's nice that pydantic doesn't have issues with mutable defaults.
